quickfixj-core/src/main/java/quickfix/field/converter/CharArrayConverter.java

```diff
@@ -0,0 +1,40 @@
+package quickfix.field.converter;
+
+import quickfix.FieldConvertError;
+
+/**
+ * Converts between character array and string.
+ */
+public class CharArrayConverter {
+
+    private static final String CHAR_ARRAY_REGEX = "[A-Za-z0-9]( [A-Za-z0-9])*";
```

Comment:
I think this regex needs to be changed to comply to the definition as per the FIX spec of `character` since it would not match punctuation: https://fiximate.fixtrading.org/en/FIX.5.0SP2_EP255/fix_datatypes.html

> Single character value, can include any alphanumeric character or punctuation except the delimiter. All char fields are case sensitive (i.e. m != M).

TBH there seems to be no field which has a char value which would not match that regex. However, I think we should adhere to the spec here.

Reply:
"Single character value, can include any alphanumeric character or punctuation except the delimiter. All char fields are case sensitive (i.e. m != M)."

It would make sens that a char is any ascii char at the end, except the | delimiter of course...

Reply:
Have a look at the latest commit guys. I added more specific test cases and I think that's what you both guys proposing.

I kept the separator to single space character because `quickfix.DataDictionary#isFieldValue` supports it.